crypto/evp/keymgmt_lib.c

```diff
@@ -217,6 +245,105 @@ void *evp_keymgmt_fromdata(EVP_PKEY *target, EVP_KEYMGMT *keymgmt,
     return provdata;
 }
 
+int evp_keymgmt_is(const EVP_PKEY *pkey, int domainparams)
+{
+    EVP_KEYMGMT *keymgmt = pkey->pkeys[0].keymgmt;
+
+    if (keymgmt == NULL)
+        return 0;          /* There's no one to ask, so assume "no" */
+    return pkey->pkeys[0].domainparams
+        ? evp_keymgmt_isdomparams(keymgmt, pkey->pkeys[0].provdata)
+        : evp_keymgmt_iskey(keymgmt, pkey->pkeys[0].provdata);
+}
+
+/*
+ * This function adheres to EVP_PKEY_cmp() / EVP_PKEY_cmp_parameters()
+ * return values, i.e.:
+ *
+ *  1   same key
```

Comment:
I really dislike this being called _cmp()
cmp() to me means return 0 if equal.

Reply:
The public name will not change, obviously, but feel free to suggest a better name for this function.

Reply:
equals() ?